datafusion/expr-common/src/type_coercion/binary.rs

```diff
@@ -2250,6 +2291,23 @@ mod tests {
         ); // nullable because the RHS is nullable
     }
 
+    #[test]
+    fn test_struct_coercion() {
```

Comment:
The basic idea looks good -- I think we would need to add some sqllogictests that show what we want to happen in order to evaluate any actual code solution

Sorry for the delay @TheBuilderJR